include/path_tracking_pid/controller.hpp

```diff
@@ -213,16 +220,17 @@ class Controller : private boost::noncopyable
   ControllerState controller_state_;
 
   // Global Plan variables
-  std::vector<tf2::Transform> global_plan_tf_;     // Global plan vector
-  std::vector<double> distance_to_goal_vector_;    // Vector with distances to goal
+  std::vector<tf2::Transform> global_plan_tf_;  // Global plan vector
+  std::vector<boost::units::quantity<boost::units::si::length>>
+    distance_to_goal_vector_;                      // Vector with distances to goal
   std::vector<double> turning_radius_inv_vector_;  // Vector with computed turning radius inverse
-  double distance_to_goal_ = 0;
+  boost::units::quantity<boost::units::si::length> distance_to_goal_{0};
   tf2::Transform current_goal_;
   tf2::Transform current_pos_on_plan_;
   tf2::Transform current_with_carrot_;
 
   // Auxiliary variables
-  double current_target_x_vel_ = 0.0;
+  boost::units::quantity<boost::units::si::velocity> current_target_x_vel_{0};
```

Comment:
I think these kind of constructs can be much more readable with a typedef in the header for the used units in this package.
```
using velocity_t = boost::units::quantity<boost::units::si::velocity>

...

velocity_t current_target_x_vel_{0};
```

Reply:
I completely agree. Can easily be done if/when we decide to use Boost units.